dev-journey-blog.md

```diff
@@ -0,0 +1,89 @@
+
+
+### Acme Freight + serverless
+
+Acme Freight uses a serverless approach to respond to weather events. An action is triggered by a storm event; retailers affected by the weather event are identified; their stock and shipments are retrieved; recommendations are made based on the storm, stock and shipments. While the guts of the code are abstracted out of this flow into their own function blocks, you can see the process in the following code from the recommend action:
+
+**`actions/recommend.js`**
+
+```javascript
+    return new Promise((resolve, reject) => {
+      async.waterfall([
+
+        // delete existing recommendations for this demo
+        function(callback) {
+          self.cleanup(
+            args['services.cloudant.url'],
+            args['services.cloudant.database'],
+            args.demoGuid,
+            callback);
+        },
+
+        // retrieve list of retailers
+        function(callback) {
+          self.getRetailers(args['services.controller.url'],
+            args.demoGuid, callback);
+        },
+
+        // identify retailers affected by the weather event
+        function(retailers, callback) {
+          self.filterRetailers(retailers, args.event, callback);
+        },
+
+        // retrieve their stock and make new shipments
+        function(retailers, callback) {
+          self.recommend(retailers, callback);
+        },
+
+        // persist the recommendations
+        function(recommendations, callback) {
+          self.persist(
+            args['services.cloudant.url'],
+            args['services.cloudant.database'],
+            args.demoGuid,
+            recommendations,
+            callback);
+        }
+      ], (err, result) => {
+        if (err) {
+          reject({ ok: false });
+        }
+        else {
+          resolve({
+            demoGuid: args.demoGuid,
+            event: args.event,
+            recommendations: result,
+          });
+        }
+      });
+    });
+```
+
+The code above has comments inline to describe the flow of the action
+
+1. delete existing recommendations for this demo
+1. retrieve list of retailers
+1. identify retailers affected by the weather event
+1. retrieve their stock and make new shipments
+1. persist the recommendations
+
+And the last code block has an if/else that either rejects the promise **if** there is an error or **else** it resolves the promise with an object of recommendation information.
+
+If you look at the actions in the repo, you'll see the functional code is broken into logical separations of concern. Then, the recommendations interface is made available to our application via APIs that are native to the OpenWhisk framework on Bluemix. Separating the recommendations logic from the Acme Freight application cuts our costs, removes deployment dependencies and frees up our developers to focus on the core application needs.
```

Comment:
add call to action to learn more about deploying an openwhisk app in bluemix - make sure we are pitching the API side of this as much as the serverless side 

should we think about adding a section about securing the openwhisk api's to tie in the bluemix native side? @svennam92 @joesepi